### ISSUE TITLE
Accessibility-> Find a site page filter expansion takes the user to the next next filters section

### ISSUE THREAD

**Issue by manuji** [bcgov/nr-rec-resources#593]
**Describe the bug**
When using the keyboard and navigating through the Filters of the Find a site or trail page, expanding to show all the filters in a section, it takes the user to the next filter section instead of taking the user the next immediate filter option on the same section. This issue fails the [2.4.3 Focus Order (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/focus-order) criterion

**To Reproduce**
1. Go to 'https://d1vrz2gt4nmfe5.cloudfront.net/search'
2. Using the keyboard, try navigating through the filters section of the page
3. Expand the filters sections by selecting the link "Show all xx" 
4. Tab again and check where the next tabbing takes the user to. 

**Expected behavior**
When navigating through content, the elements should receive focus in an order that preserves meaning. 

**Screenshots**

https://github.com/user-attachments/assets/e3ab2e66-cc5b-498b-be3b-dfbe85f9e466
